net/miniupnpd/files/miniupnpd.hotplug

```diff
@@ -12,18 +12,35 @@
 [ "$ACTION" != "ifup" ] && procd_running "miniupnpd" "*" && exit 0
 
 tmpconf="/var/etc/miniupnpd.conf"
-extiface=$(uci get upnpd.config.external_iface)
-extzone=$(uci get upnpd.config.external_zone)
+external_iface=$(uci get upnpd.config.external_iface)
+external_iface6=$(uci get upnpd.config.external_iface6)
+external_zone=$(uci get upnpd.config.external_zone)
 
 . /lib/functions/network.sh
 
-[ -z "$extiface" ] && {
-  # manual external zone (if dynamically find interfaces
-  # belonging to it) overrides network_find_wan*
-  [ -n "$extzone"  ] && ifname=$(fw3 -q zone "$extzone" | head -1)
-  [ -z "$extiface" ] && network_find_wan extiface
-  [ -z "$extiface" ] && network_find_wan6 extiface
-}
-
-[ -z "$ifname" ] && network_get_device ifname "$extiface"
-grep -q "ext_ifname=$ifname" "$tmpconf" || /etc/init.d/miniupnpd restart
+if [ -n "$external_iface" ] ; then
+	network_get_device ifname "$external_iface"
+else
+	if [ -n "$external_zone" ] ; then
+		ifname=$(fw3 -q zone "$external_zone" | head -1)
+	else
+		network_find_wan external_iface
```

Comment:
The `network_find_wan` function might fail, I suggest chaining the calls:

`network_find_wan external_iface && network_get_device ifname "$external_iface"`

Reply:
Done - something else the original script didn't do ;-)